src/utils/generate_numbering.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { INumberingDictionary } from './numbering_dictionary';
+
+// Maximum heading level:
+const MAX_HEADING_LEVEL = 6;
+
+/**
+ * Updates numbering dictionary levels.
```

Comment:
Can we say in what way this updates it?

Reply:
TBH I never really understood the intent behind the original logic or why mutation was necessary. I think the main rationale for this being in a separate function in the original codebase was to circumvent compiler errors related to setting levels to non-numeric values. But this is only a guess.

Nor did I trace how this mutation might be depended on throughout the application.

So, I punted on this, opting for a (less than optimal) generic description, which I think is fine, for this particular case, as it is an internal helper function which simply encapsulates some logic.